### ISSUE TITLE
py-Add hyperparameter optimization to modeltrainer

### ISSUE THREAD

**Issue by davidwhealey** [HealthCatalyst/healthcareai-py#72]
In the supervised learning model trainer, currently we're just training a model and doing cross validation for stats.  We should implement a grid search. 

Note: This will also mean we should create a separate test set for getting the model stats.
